### DISCUSSION TITLE
Deploy Error

### DISCUSSION THREAD

**Discussion by Acacio98** [InterDigitalInc/AdvantEDGE#361] [Q&A]
Hello, when installing the platform, it gives an error when initializing "kubeadm", as it has again released a new version of Docker and gives the following error: 

~$ sudo kubeadm init
I0606 17:53:31.293296	4739 version.go:252] remote version is much newer: v1.24.1; falling back to: stable-1.19
W0606 17:53:31.779612	4739 configset.go:348] WARNING: kubeadm cannot validate component configs for API groups [kubelet.config.k8s.io kubeproxy.config.k8s.io]
[init] Using Kubernetes version: v1.19.16
[preflight] Running pre-flight checks
    [WARNING SystemVerification]: this Docker version is not on the list of validated versions: 20.10.16. Latest validated version: 19.03
error execution phase preflight: [preflight] Some fatal errors occurred:
    [ERROR NumCPU]: the number of available CPUs 1 is less than the required 2
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher

**Comment by dilallkx** [comment]
Hi @Acacio98,

Thank you for reaching out!

Your issue seems to be related to the kubernetes server version installed by kubeadm during init. The installed server version is the latest k8s patched version available at the time kubeadm init is run, which explains some possible mismatches between the k8s client & server versions when you run the command `kubectl version`. If you want to install an older patched version you can do so using the following command during kubeadm init (https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/), for example:

```
sudo kubeadm init --kubernetes-version="v1.19.14"
```

Simply find & install the version that previously worked on your setup.

A few additional notes:
- The Docker logs are just a warning and do not seem to cause issues on our end; we have successfully run with the most recent Docker updates.
- Kubeadm recommends at least 2 cpus (https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/); anything less could cause issues.
- You can probably add a flag to ignore the preflight errors, although we have not tested this

Hope this helps!
Regards!

**Reply by Acacio98** [reply]
Hi again @dilallkx 

I have tried to solve the problem using the command described in your reply but I keep getting the same error. I have been getting failure since the new Docker update. I also don't know why it requires 2 CPUs if the first times I ran it on my laptop it worked perfectly. What can I do? I need to fix it soon.

Thanks

**Reply by dilallkx** [reply]
Hi @Acacio98,

The command I proposed installs v1.19.14, which may not be the version to need. I would recommend trying an older version that you know previously worked on your setup.

If the problem persists you can then try downgrading the docker version using a procedure like this:

**NOTES:**
- You should stop & uninstall AdvantEDGE & k8s before reinstalling docker
- You will need to rebuild & dockerize the platform after reinstalling docker
- You should use the previously working version of docker in the docker installation command below

```
# Uninstall the Docker Engine, CLI, Containerd, and Docker Compose packages
sudo apt-get purge docker-ce docker-ce-cli containerd.io docker-compose-plugin

# Delete all images, containers, and volumes
sudo rm -rf /var/lib/docker

sudo reboot

# Install dependencies
sudo apt-get update
sudo apt-get install ca-certificates curl gnupg lsb-release

# Add Docker’s official GPG key
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg

# Set up the stable repository
echo \
  "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null

# Install Docker engine
sudo apt-get update
sudo apt-get install -y docker-ce=5:20.10.14~3-0~ubuntu-bionic docker-ce-cli=5:20.10.14~3-0~ubuntu-bionic containerd.io=1.5.11-1 docker-compose-plugin

# Lock current version
sudo apt-mark hold docker-ce docker-ce-cli containerd.io docker-compose-plugin

# Add user to docker group
sudo usermod -aG docker <your-user>

# Restart shell to apply changes
```

I hope this helps resolve your issues.
Regards!

**Reply by Acacio98** [reply]
Hi @dilallkx , 
When I enter the following command: (sudo apt-get install -y docker-ce=5:20.10.14~3-0~ubuntu-bionic docker-ce-cli=5:20.10.14~3-0~ubuntu-bionic containerd.io=1.5.11-1 docker-compose-plugin) I get that it is not possible to find that version, I enter the command to see the list of available versions and I do not see any, what could be wrong? 

Thank you very much for your answer.

**Reply by dilallkx** [reply]
Hi @Acacio98,

Apologies for the delayed response!
Have you had any success downgrading docker on your setup?
Note that you can obtain the available versions using the command: `apt-cache madison docker-ce`
I hope this helps!

Regards.

**Reply by Acacio98** [reply]
Hi @dilallkx ,
I have started to install the project from scratch. I changed the version of Docker to an older one like 19.03.15, with the following command: (sudo apt-get install -y docker-ce=5:19.03.15~3-0~ubuntu-focal docker-ce-cli=5:19.03.15~ubuntu-focal containerd.io=1.5. 11-1 docker-compose-plugin), then I installed kubernetes with the command that appears in AdvantEdge, when I launch the command: (sudo kubeadm init), I get the same error, I also tried to run the following command as you told me: (sudo kubeadm init --kubernetes-version="v1.19.14"
) and I still get the same error. I need it to work as soon as possible because I'm doing a project for the university and it's taking too long, I don't know how to solve the problem. 

Thanks for the answer. 
![Captura de pantalla de 2022-07-04 19-21-12](https://user-images.githubusercontent.com/98397805/177197899-a18c354a-eb92-4d8e-9e6d-2514e0fc9bb3.png)

**Reply by dilallkx** [reply]
Hi @Acacio98,

Thank you for providing your latest results.
I provided an example command that installs v1.19.14. It seems this version also has the preflight CPU check. I recommend you try the following options:
- Install the exact version of k8s that was previously working on your setup (e.g. 1.19.1?)
- Install on a system with at least 2 CPUs
- Disable preflight CPU check (https://stackoverflow.com/questions/60804280/kubadm-init-error-cpus-1-is-less-than-required-2)

I hope this helps resolve your issue.
Regards!
